Sources/PhoenixLiveViewNative/LiveViewCoordinator.swift

```diff
@@ -479,8 +481,10 @@ public class LiveViewCoordinator<R: CustomRegistry>: ObservableObject {
         // todo: what should happen if decoding or parsing fails?
         self.rendered = try! Root(from: FragmentDecoder(data: renderedPayload))
         let elements = try! self.parseDOM(html: self.rendered.buildString(), baseURL: self.currentURL)
-        self.internalState = .connected
-        self.elements = elements
+        DispatchQueue.main.async {
+            self.internalState = .connected
+            self.elements = elements
+        }
```

Comment:
`handleJoinPayload` should always be called on the main thread, so this should be unnecessary. But, the call to it in `setupChannelJoinHandlers` (line 443) needs to be wrapped in a `Task { @MainActor in }`.